### PR TITLE
Refactor udf_percentile in telemetry histogram percentiles as a persistent udf 

### PR DESCRIPTION
**Pull request by acmiyaguchi**
The user defined function for calculating histogram percentiles is complex enough to warrant tests. I have been adapting this function for calculating percentiles for #829, but I'm not sure how to determine whether the results look sane. By moving the reference udf into it's own module, it can be unit-tested. This should also determine whether #831 is actually an issue.

I've decided to leave the logic as is, and document a few test cases with what I expect the values to be and what the actual value is for a few medians.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
@relud Is this the right way to factor out a javascript udf from the sql source, or does this have to be done in two separate PRs to make dryrun succeed?

**Comment by relud** [comment]
> @relud Is this the right way to factor out a javascript udf from the sql source

yes

> does this have to be done in two separate PRs to make dryrun succeed?

to make the dry run succeed without multiple PRs: wait for this to get approved, manually publish the UDF, manually re-run the failed dry-run-sql in CI

File: tests/assert/approx_equals.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        'Expected ',
+        TO_JSON_STRING(ROUND(expected, 2)),
+        ' ± ',
+        TO_JSON_STRING(tolerance),
+        ' but got ',
+        TO_JSON_STRING(ROUND(actual, 2))
+      )
+    )
+  )
+);
```
Comment (relud) [inline tests/assert/approx_equals.sql]:
```suggestion
);

-- Tests
SELECT
  assert_approx_equals(1, 1, 0),
  assert_approx_equals(1, 2, 1);
  
#xfail
SELECT
  assert_approx_equals(1,2,0.9);
```

Comment (acmiyaguchi) [reply tests/assert/approx_equals.sql]: Ah, I should have figured that you could write tests for the test helpers. TIL about the `xfail` directive in the testing framework, that'll come in handy :+1:.

File: udf_js/telemetry_percentile.sql [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  }
+  percentileBucketIndex--;
+
+  //Undo the last loop iteration where we overshot
+  hitsAtPercentileInBar += histogram[percentileBucketIndex].value;
+
+  // The ratio of the hits in the percentile to the hits in the bar containing it.
+  // How far we are inside the bar?
+  let ratioInBar = hitsAtPercentileInBar / histogram[percentileBucketIndex].value;
+  if (type === "histogram-exponential") {
```
Comment (emtwo) [inline udf_js/telemetry_percentile.sql]: thank you!

**Pull request merged**